### PR TITLE
Refactored integration tests

### LINKED ISSUE #21: Handle exceptions in integration tests
## Motivation

In integration tests exceptions are only handled and verified that they were thrown. We don't actually differentiate one from another, and don't assert on the actual message content.

## Implementation note
- Consider looking at [`tests/integration/arg_test.py`](https://github.com/candy-kingdom/cli/blob/develop/tests/integration/arg_test.py) as an example of how exceptions are checked.

### PR DESCRIPTION
**Pull request by JoshuaLight**
Closes #21.

### CODE REVIEW

File: candies/cli/parsers/standard.py [hunk 1]
```diff
@@ -88,6 +88,11 @@ def prefixes(signature: inspect.Signature) -> str:
     def prefix_of(x: inspect.Parameter) -> str:
         prefix = getattr(x.annotation, 'prefix', '-')
 
+        if prefix == None:
```
Comment (konstantin-ogulchansky) [inline candies/cli/parsers/standard.py:91]: I think we should use `is` for comparing with `None` or `True`/`False`.

File: candies/cli/parsers/standard.py [hunk 2]
```diff
@@ -88,6 +88,11 @@ def prefixes(signature: inspect.Signature) -> str:
...
+            raise ValueError(f"Argument '{x.name}' has an invalid prefix "
+                             f"'{prefix}': "
```
Comment (konstantin-ogulchansky) [inline candies/cli/parsers/standard.py:93]: Could we concatenate `f"'{prefix}': "` and `f"it is None"` and write them on the same line?

File: tests/execution.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        with redirect(output):
+            result = execute_()
+
+        return result, output.getvalue()
+
+    def execute_():
+        try:
+            return cli(with_.split())
+        # `BaseException`, because `argparse` throws `SystemExit` on error.
```
Comment (konstantin-ogulchansky) [inline tests/execution.py:42]: Is this comment still relevant?

File: candies/cli/parsers/standard.py (hunk 1 above)
Comment (JoshuaLight) [reply candies/cli/parsers/standard.py:91]: Fixed!

File: candies/cli/parsers/standard.py (hunk 2 above)
Comment (JoshuaLight) [reply candies/cli/parsers/standard.py:93]: I wanted the string to be symmetric to other strings in the function: both are 3 lines tall. ))

File: tests/execution.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    Supports redirection of either `sys.stdout` or `sys.stderr` for testing
+    purposes.
+
+    Args:
+        cli: A callable that represents a CLI function.
+        with_: A string that represents a command line.
+        redirect: An object that redirects an output into file-like object
+            using context manager protocol.
+
+    Examples:
```
Comment (JoshuaLight) [inline tests/execution.py:21]: I think we need `Returns` documentation here.

**Pull request merged**